File: apis/elysium-core-api/src/main/java/com/seventh_root/elysium/core/util/MathUtils.java
package com.seventh_root.elysium.core.util;

public class MathUtils {

    private MathUtils() {}

    public static double fastSqrt(double number) {
        return Double.longBitsToDouble(((Double.doubleToLongBits(number) - (1L << 52)) >> 1) + (1L << 61));
    }

}

File: apis/elysium-core-api/src/main/kotlin/com/seventh_root/elysium/core/util/MathUtils.kt
package com.seventh_root.elysium.core.util

object MathUtils {

    fun fastSqrt(number: Double): Double {
        return java.lang.Double.longBitsToDouble((java.lang.Double.doubleToLongBits(number) - (1L shl 52) shr 1) + (1L shl 61))
    }

}
